### PR TITLE
UHF-9050 button refactor

### PR DESCRIPTION
**Pull request by Arkkimaagi**
# [UHF-9050](https://helsinkisolutionoffice.atlassian.net/browse/UHF-9050)

## What was done

* This thing was fixed

## How to install

* Make sure your instance is up and running on latest dev branch.
  * `git pull origin dev`
  * `make fresh`
* Update the HDBT theme
  * `composer require drupal/hdbt:dev-UHF-9050_button_refactor`
* Run `make drush-cr`

## How to test

* [ ] Check that this feature works
* [ ] Check that code follows our standards

## Designers review

* [ ] This PR does not need designers review
* [ ] This PR has been visually reviewed by a designer (Name of the designer)

## Other PRs

* Link to other PR

[UHF-9050]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-9050?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/scss/05_objects/_button.scss [hunk 1]
```diff
@@ -132,208 +178,272 @@ input[type='submit'].hds-button {
...
+input[type='submit'].hds-button--small {
+  padding: var(--spacing-2-xs) var(--spacing-m);
+}
+
+/* both icons */
+.hds-button--small .hds-button__label:not(:first-of-type,:last-of-type) {
+  padding: var(--spacing-2-xs) var(--spacing-2-xs);
+}
+
+/* FULL WIDTH */
```
Comment (teroelonen) [inline src/scss/05_objects/_button.scss:231]: Change to one-line-comment style.

File: src/scss/05_objects/_button.scss [hunk 2]
```diff
@@ -132,208 +178,272 @@ input[type='submit'].hds-button {
...
-/* ICONS */
+// SMALL
+.hds-button--small {
+  padding: 0;
+}
+
+input[type='button'].hds-button--small,
+input[type='reset'].hds-button--small,
+input[type='submit'].hds-button--small {
+  padding: var(--spacing-2-xs) var(--spacing-m);
```
Comment (teroelonen) [inline src/scss/05_objects/_button.scss:223]: This is overriding styles set on line 191. Is this intentional?

**Review by teroelonen — CHANGES_REQUESTED** [review]
Amazing work with this! Really good stuff! Just one question and small notice there on the button.scss file :)

File: src/scss/05_objects/_button.scss (hunk 2 above)
Comment (Arkkimaagi) [reply src/scss/05_objects/_button.scss:223]: This part of our code tries to closely follow HDS codes. This part is in https://github.com/City-of-Helsinki/helsinki-design-system/blob/76ed6122f96cfe847b0ff94c13ea0afd9e7cc067/packages/core/src/components/button/button.css#L233C15-L233C15 and has a same problem. So kind of intentional on our end, but do not know how intentional is it is on their end.

File: src/scss/05_objects/_button.scss (hunk 1 above)
Comment (Arkkimaagi) [reply src/scss/05_objects/_button.scss:231]: Oops, right, these must have slipt past when changing HDS CSS comments to SCSS comments on our side

**Comment by khalima** [comment]
Closing this PR as the same feature and changes was merged in https://github.com/City-of-Helsinki/drupal-hdbt/pull/810

**Pull request closed without merging**